lib/jsonapi-response.js

```diff
@@ -45,46 +45,68 @@ function getRelationships (data) {
           id: ''
         };
 
-        rel.id = TypeMapping.toExternal(el.admin.uid);
-        rel.type = TypeMapping.toExternal(el.admin.id.split('-')[0]);
-
-        relationships[key].data.push(rel);
+        if (el.admin) {
+          rel.id = TypeMapping.toExternal(el.admin.uid);
+          rel.type = TypeMapping.toExternal(el.admin.id.split('-')[0]);
+          relationships[key].data.push(rel);
+        } else {
+          addAsAttribute(el, key, response);
+        }
       });
 
       if (relationships[key].length === 1) {
         relationships[key] = relationships[key][0];
       }
+
+      if (!relationships[key].data.length) {
+        delete relationships[key];
+      }
     }
   });
 
-  return relationships;
+  const creationDetails = getCreationDetails(data, config);
```

Comment:
In this file we seem to be augmenting objects passed to the functions, but this one is different and returns an object to be merged in.

In general we should try to create functions that operate on parameters and return a value. Augmenting objects couples the function to the data passed in and will cause it to make assumptions about its structure that you can't guarantee or always guard against in every case. Augmenting is what's commonly known as a side effect which should be avoided as much as possible to make your code more robust and testable.

It's up to you how you want to handle this, it might be an idea to just be consistent and pass the relationships in to be augmented.

As a comparison, the search results has a `createRelationships` function that returns an object with `relationships` and `included` properties - https://github.com/TheScienceMuseum/collectionsonline/blob/master/lib/transforms/search-results-to-jsonapi.js#L68